src/main/java/net/mcreator/ui/minecraft/TextureComboBox.java

```diff
@@ -128,6 +128,11 @@ public String getTextureName() {
 		return selected.getTextureName() + (addPNGExtension ? ".png" : "");
 	}
 
+	public void changeTextureType(TextureType newType) {

```

Comment:
This is not ok as currently in GeneratableElement, annotation only marks one texture type for use with the references system. If you need different texture type, it is best to provde a separate selector and separate object field in GE object